spec/lib/gateway/stat_gateway_spec.rb

```diff
@@ -0,0 +1,126 @@
+describe StatGateway do
+  before do
+    DB[:userdetails].truncate
+  end
+
+  context 'given no signups' do
+    it 'returns zero signups for today' do
+      expect(subject.signups[:today]).to eq(0)
+    end
+
+    it 'returns zero signups for today' do
+      expect(subject.signups[:total]).to eq(0)
+    end
+  end
+
+  context 'given 3 signups today' do
+    before do
+      3.times do |i|
+        User.create(username: "Today #{i}")
+      end
+    end
+
+    it 'returns signups for today' do
+      expect(subject.signups[:today]).to eq(3)
+    end
+
+    it 'returns same total number of signups' do
+      expect(subject.signups[:total]).to eq(3)
+    end
+  end
+
+  context 'given 5 signups today and 1 yesterday' do
+    before do
+      User.create(username: 'Yesterday', created_at: Date.today - 1)
+
+      5.times do |i|
+        User.create(username: "Today #{i}")
+      end
+    end
+
+    it 'returns zero signups 5 signups for today' do
+      expect(subject.signups[:today]).to eq(5)
+    end
+
+    it 'returns zero signups 6 signups total' do
+      expect(subject.signups[:total]).to eq(6)
+    end
+  end
+
+  context 'given 2 signups tomorrow' do
+    before do
+      2.times do |i|
+        User.create(username: "Tomorrow #{i}", created_at: Date.today + 1)
+      end
+    end
+
+    it 'returns zero signups for today' do
+      expect(subject.signups[:today]).to eq(0)
+    end
+
+    it 'returns zero signups total' do
+      expect(subject.signups[:today]).to eq(0)
+    end
+  end
+
+  context 'given 1 SMS signup today' do
```

Comment:
Given 1 sms signup and 1 email signup?

I'd then also be explicit around setting the contact field.